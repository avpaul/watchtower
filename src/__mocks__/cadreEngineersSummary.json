{
  "current_page": 1,
  "data": [
    {
      "id": 1,
      "fellow_id": "-LSKaKObA5T-XFQIUEKV",
      "first_name": "Chidozie",
      "last_name": "Nwoga",
      "email": "chidozie.nwoga@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?39044",
      "cohort": "KLA-1",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-10-31 00:00:00",
      "sims_end_date": "2019-01-10 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Duncan Ireri",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Javascript/Ruby",
      "apprenticeship_start_date": "2019-06-11 00:00:00",
      "apprenticeship_end_date": "2019-06-26 00:00:00",
      "apprenticeship_manager": "Njenga Njunge",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": "TC"
    },
    {
      "id": 2,
      "fellow_id": "-LC8zTmDnKCIjiXf29-y",
      "first_name": "Ronald",
      "last_name": "Mutebi",
      "email": "ronald.mutebi@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?13109",
      "cohort": "KLA-2",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-10-22 00:00:00",
      "sims_end_date": "2019-02-21 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Victoria Offoma",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Javascript/Ruby",
      "apprenticeship_start_date": "2019-06-05 00:00:00",
      "apprenticeship_end_date": "2019-06-20 00:00:00",
      "apprenticeship_manager": "Njenga Njunge",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": "TC"
    },
    {
      "id": 3,
      "fellow_id": "-LONlfPRQ9t1dFdgu2y8",
      "first_name": "Brian",
      "last_name": "Bett",
      "email": "brian.bett@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?43935",
      "cohort": "KLA-3",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-11-07 00:00:00",
      "sims_end_date": "2019-01-05 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Victoria Offoma",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Javascript/Ruby",
      "apprenticeship_start_date": "2019-06-07 00:00:00",
      "apprenticeship_end_date": "2019-06-29 00:00:00",
      "apprenticeship_manager": "Njenga Njunge",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": "TC"
    },
    {
      "id": 4,
      "fellow_id": "-L9BGHvLBPPIAXMwx5kX",
      "first_name": "Collins",
      "last_name": "Muru",
      "email": "collins.muru@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?13109",
      "cohort": "KLA-4",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-11-02 00:00:00",
      "sims_end_date": "2019-01-17 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Victoria Offoma",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Javascript/Ruby",
      "apprenticeship_start_date": "2019-06-04 00:00:00",
      "apprenticeship_end_date": "2019-06-15 00:00:00",
      "apprenticeship_manager": "Njenga Njunge",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": "TC"
    },
    {
      "id": 5,
      "fellow_id": "-LQcqbQzyNpIlfJreeiZ",
      "first_name": "Baker",
      "last_name": "Sekitoleko",
      "email": "baker.sekitoleko@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475",
      "cohort": "KLA-5",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-10-28 00:00:00",
      "sims_end_date": "2019-02-05 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Martin Baale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Javascript/Ruby",
      "apprenticeship_start_date": "2019-06-06 00:00:00",
      "apprenticeship_end_date": "2019-06-24 00:00:00",
      "apprenticeship_manager": "Njenga Njunge",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": "TC"
    },
    {
      "id": 6,
      "fellow_id": "-LSK_lcw0_TQdDTY11ow",
      "first_name": "Peace",
      "last_name": "Oyedeji",
      "email": "peace.oyedeji@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475",
      "cohort": "KLA-6",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-10-23 00:00:00",
      "sims_end_date": "2019-01-19 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Martin Baale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Android",
      "apprenticeship_start_date": "2019-06-10 00:00:00",
      "apprenticeship_end_date": "2019-06-16 00:00:00",
      "apprenticeship_manager": "Roger Taracha",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 7,
      "fellow_id": "-LLUploBG_-tXDaBuxzb",
      "first_name": "Brian",
      "last_name": "Ashiundu",
      "email": "brian.mboya@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?39044",
      "cohort": "KLA-7",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-11-08 00:00:00",
      "sims_end_date": "2019-02-16 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Martin Baale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Android",
      "apprenticeship_start_date": "2019-06-13 00:00:00",
      "apprenticeship_end_date": "2019-06-14 00:00:00",
      "apprenticeship_manager": "Roger Taracha",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 8,
      "fellow_id": "-LQdd1UDOZec3bdQzH2O",
      "first_name": "Eric",
      "last_name": "Nyokabi",
      "email": "erick.nyokabi@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?35451",
      "cohort": "KLA-8",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-11-09 00:00:00",
      "sims_end_date": "2019-01-30 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Brian Maiyo",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Android",
      "apprenticeship_start_date": "2019-06-09 00:00:00",
      "apprenticeship_end_date": "2019-06-19 00:00:00",
      "apprenticeship_manager": "Roger Taracha",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 9,
      "fellow_id": "-LLUW2BwQiZS0j9KF10j",
      "first_name": "Fred",
      "last_name": "Adegoke",
      "email": "fred.adegoke@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?13109",
      "cohort": "Lagos-1",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Django",
      "sims_start_date": "2018-10-21 00:00:00",
      "sims_end_date": "2019-02-25 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Brian Maiyo",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Android",
      "apprenticeship_start_date": "2019-06-03 00:00:00",
      "apprenticeship_end_date": "2019-06-21 00:00:00",
      "apprenticeship_manager": "Roger Taracha",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 10,
      "fellow_id": "-LQZkSBEZi5DQ-J4yQ-9",
      "first_name": "Ibrahim",
      "last_name": "Alausa",
      "email": "ibrahim.alausa@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?35451",
      "cohort": "Lagos-2",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-10-20 00:00:00",
      "sims_end_date": "2019-01-11 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Brian Maiyo",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-12 00:00:00",
      "apprenticeship_end_date": "2019-06-28 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 11,
      "fellow_id": "-LLUpLy53gzYC02PhXK2",
      "first_name": "Lazarus",
      "last_name": "Odhiambo",
      "email": "lazarus.odhiambo@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?39044",
      "cohort": "Lagos-3",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-10-19 00:00:00",
      "sims_end_date": "2019-01-16 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Brian Maiyo",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-08 00:00:00",
      "apprenticeship_end_date": "2019-06-27 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 12,
      "fellow_id": "-LLUWJwQ6jw2xwXpZreD",
      "first_name": "Oluseyi",
      "last_name": "Anani",
      "email": "oluseyi.anani@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?13109",
      "cohort": "Lagos-4",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-03 00:00:00",
      "sims_end_date": "2019-01-23 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-02 00:00:00",
      "apprenticeship_end_date": "2019-06-23 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 13,
      "fellow_id": "-LQdbZPS_yMPuO8LjGjC",
      "first_name": "Paul",
      "last_name": "Gichuki",
      "email": "paul.gichuki@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?43935",
      "cohort": "Lagos-5",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-10-26 00:00:00",
      "sims_end_date": "2019-02-20 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-12 00:00:00",
      "apprenticeship_end_date": "2019-06-22 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 14,
      "fellow_id": "-LQdcYBwXM9khXkSHg4I",
      "first_name": "Steven",
      "last_name": "Otieno",
      "email": "steven.otieno@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?13109",
      "cohort": "Lagos-6",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-10 00:00:00",
      "sims_end_date": "2019-01-09 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-06 00:00:00",
      "apprenticeship_end_date": "2019-06-25 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 15,
      "fellow_id": "-LSKa7Lv0YJtaXSlGbog",
      "first_name": "Wasibani",
      "last_name": "Roy",
      "email": "roy.wasibani@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475",
      "cohort": "Lagos-7",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-12 00:00:00",
      "sims_end_date": "2019-02-17 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-02 00:00:00",
      "apprenticeship_end_date": "2019-06-17 00:00:00",
      "apprenticeship_manager": "Jason Okagbare",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 16,
      "fellow_id": "-LP5MeqRJFJQGKwfG1vr",
      "first_name": "Lindsey",
      "last_name": "Patra",
      "email": "lindsey.patra@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?35451",
      "cohort": "Lagos-8",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-13 00:00:00",
      "sims_end_date": "2019-02-18 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "PHP/Lumen",
      "apprenticeship_start_date": "2019-06-03 00:00:00",
      "apprenticeship_end_date": "2019-06-18 00:00:00",
      "apprenticeship_manager": "Olaolu Akinsete",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 17,
      "fellow_id": "-LQZkSO0EbLk96oQmz34",
      "first_name": "Didunloluwa ",
      "last_name": "Morodolu",
      "email": "didunloluwa.morodolu@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475",
      "cohort": "NBO-9",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-14 00:00:00",
      "sims_end_date": "2019-02-19 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "PHP/Lumen",
      "apprenticeship_start_date": "2019-06-04 00:00:00",
      "apprenticeship_end_date": "2019-06-19 00:00:00",
      "apprenticeship_manager": "Olaolu Akinsete",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 18,
      "fellow_id": "-LP5MeqRJFJQGKwfG1vd",
      "first_name": "Ivan",
      "last_name": "Rutale",
      "email": "ivan.rutale@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475 ",
      "cohort": "NBO-10",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2018-11-14 00:00:00",
      "sims_end_date": "2019-02-19 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Daniel Ale",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "PHP/Lumen",
      "apprenticeship_start_date": "2019-06-04 00:00:00",
      "apprenticeship_end_date": "2019-06-04 00:00:00",
      "apprenticeship_manager": "Olaolu Akinsete",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 19,
      "fellow_id": "-LP5MeqRJFJQGKwG1vd",
      "first_name": "Morris",
      "last_name": "Njuguna",
      "email": "morris.njuguna@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475 ",
      "cohort": "NBO-11",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2019-02-19 00:00:00",
      "sims_end_date": "2019-02-19 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Olaolu Akinsete",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "PHP/Lumen",
      "apprenticeship_start_date": "2019-06-04 00:00:00",
      "apprenticeship_end_date": "2019-06-04 00:00:00",
      "apprenticeship_manager": "Olaolu Akinsete",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    },
    {
      "id": 20,
      "fellow_id": "-LP5MeqJFJQGKwG1vd",
      "first_name": "Vincent",
      "last_name": " Odhiambo",
      "email": "vincent. odhiambo@andela.com",
      "picture": "https://lorempixel.com/100/100/people/?23475",
      "cohort": "NBO-12",
      "sims_project": "AuthorsHaven",
      "sims_project_technology": "Python/Go",
      "sims_start_date": "2019-10-19 00:00:00",
      "sims_end_date": "2019-12-19 00:00:00",
      "cadre_start_date": null,
      "sims_manager": "Olaolu Akinsete",
      "apprenticeship_project": "WatchTower",
      "apprenticeship_technology": "Devops",
      "apprenticeship_start_date": "2019-06-12 00:00:00",
      "apprenticeship_end_date": "2019-06-05 00:00:00",
      "apprenticeship_manager": "Olaolu Akinsete",
      "account_active": false,
      "email_sent": false,
      "project_id": null,
      "project_role_id": null,
      "created_at": null,
      "updated_at": null,
      "project": null,
      "role": null
    }
  ],
  "first_page_url": "http://localhost:8000/api/v2/reports/fellows?page=1",
  "from": 1,
  "last_page": 2,
  "last_page_url": "http://localhost:8000/api/v2/reports/fellows?page=2",
  "next_page_url": "http://localhost:8000/api/v2/reports/fellows?page=2",
  "path": "http://localhost:8000/api/v2/reports/fellows",
  "per_page": 20,
  "prev_page_url": null,
  "to": 20,
  "total": 24
}
