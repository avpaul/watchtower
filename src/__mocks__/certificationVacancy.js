export const getCertificationVacancies = (count = 1) => {
  const vacancies = [];
  for (let i = 0; i < count; i += 1) {
    vacancies.push({
      certification: {
        id: i + 1,
        name: 'Mr. Freddie Doyle',
        description: 'Aut eaque voluptate ut eum autem autem quis.',
        exclusive: false,
        duration: 20,
        deleted_at: null
      },
      vacancy_details: {
        id: 1,
        certification_id: 1,
        is_active: false,
        created_at: '2019-08-24 22:51:15',
        updated_at: '2019-08-24 23:08:35',
        requester: 'collins.muru@andela.com',
        start_date: '2019-08-31 00:00:00',
        end_date: '2019-09-14 00:00:00',
        closing_date: '2019-08-23 00:00:00',
        fellow_id: null,
        cycle_id: i + 1,
        deleted_at: null,
        applications: []
      },
      available_slots: 2
    });
  }
  return vacancies;
};

export default getCertificationVacancies;
