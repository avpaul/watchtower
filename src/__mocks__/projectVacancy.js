export const projectVacancyMock = {
  id: 1,
  project_id: 7,
  project_role_id: 1,
  fellow_id: null,
  is_active: false
};

/**
 * Generates mock project vacancies
 *
 * @param integer count The number of vacancies to generate
 * @param integer start The id to start from
 * @return array List of vacancy objects
 */
export const generateVacancies = (count, start = 0) => {
  const vacancies = [];
  for (let index = start; index < count; index += 1) {
    vacancies.push({ ...projectVacancyMock, id: index + 1 });
  }
  return vacancies;
};

const vacanciesGroupMock = {
  project: {
    id: 7,
    name: 'Corrupti non.'
  },
  role: {
    id: 1,
    name: 'Engineer'
  }
};

/**
 * Generates mock project vacancy groups
 *
 * @param integer count The number of groups to create
 * @param integer vacancyCount The number of vacancies for each group
 * @return array List of project vacancy groups
 */
export const generateVacancyGroups = (count, vacancyCount, start = 0) => {
  const vacancies = [];
  for (let index = start; index < count; index += 1) {
    vacancies.push({
      project: {
        id: (index + 1) * 2,
        name: 'Corrupti non.'
      },
      role: {
        id: index + 1,
        name: 'Engineer'
      },
      vacancies: generateVacancies(vacancyCount),
      available_slots: vacancyCount
    });
  }
  return vacancies;
};

export default {
  ...vacanciesGroupMock,
  vacancies: generateVacancies(5),
  available_slots: 5
};
