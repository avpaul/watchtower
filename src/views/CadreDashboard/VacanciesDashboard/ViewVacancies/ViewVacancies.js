import React, { Component } from 'react';
import PropTypes from 'prop-types';
import arrayKey from 'weak-key';
import TDDOPsVacancyCard from '../../../../components/TDDOpsVacancyCard';
import './ViewVacancies.scss';

import PaginationFrontendWrapper from '../../../../components/Pagination/PaginationWrapper';

export class ViewRoleVacancies extends Component {
  state = {
    vacanciesToDisplay: 'project'
  };

  componentDidMount() {
    const {
      vacancies: { projectVacancies }
    } = this.props;
    this.updateInitialState(projectVacancies || []);
  }

  updateInitialState = vacancies => {
    const { paginationWrapper } = this.props;
    paginationWrapper.updateData(vacancies, { perPage: 20 });
  };

  updatePaginateData = toDisplay => {
    const { vacancies, paginationWrapper } = this.props;
    const vacanciesArray =
      toDisplay === 'project'
        ? vacancies.projectVacancies
        : vacancies.certificationVacancies;
    paginationWrapper.updateData(vacanciesArray || []);
  };

  toggleVacanciesToDisplay = e => {
    this.updatePaginateData(e.target.value);
    this.setState({
      vacanciesToDisplay: e.target.value
    });
  };

  renderToggleButtons = () => {
    const { vacanciesToDisplay } = this.state;
    return (
      <div className="toogle-container">
        <button
          id="project-button"
          type="button"
          value="project"
          className={
            vacanciesToDisplay === 'project'
              ? 'vacancy-toggle-button--active'
              : 'vacancy-toggle-button'
          }
          onClick={this.toggleVacanciesToDisplay}
        >
          Project Vacancies
        </button>
        <button
          id="certification-button"
          type="button"
          value="certification"
          className={
            vacanciesToDisplay === 'certification'
              ? 'vacancy-toggle-button--active'
              : 'vacancy-toggle-button'
          }
          onClick={this.toggleVacanciesToDisplay}
        >
          Certification Vacancies
        </button>
      </div>
    );
  };

  renderVacancies = (vacancies, vacanciesToDisplay) => {
    const mappedVacancies =
      vacancies.length !== 0 ? (
        vacancies.map(vacancy => (
          <TDDOPsVacancyCard
            key={arrayKey(vacancy)}
            vacancy={vacancy}
            vacanciesToDisplay={vacanciesToDisplay}
          />
        ))
      ) : (
        <div className="ops-no-vacancies">No Vacancies</div>
      );
    return mappedVacancies;
  };

  render() {
    const { paginationWrapper } = this.props;
    const { vacanciesToDisplay } = this.state;
    return (
      <div>
        <div>{this.renderToggleButtons()}</div>
        <div className="ops-vacancies-container">
          {this.renderVacancies(
            paginationWrapper.state.paginatedData,
            vacanciesToDisplay
          )}
        </div>
        <div className="">{paginationWrapper.renderPagination()}</div>
      </div>
    );
  }
}

const PaginatedVacanciesDashboard = props => (
  <PaginationFrontendWrapper component={<ViewRoleVacancies {...props} />} />
);

ViewRoleVacancies.propTypes = {
  vacancies: PropTypes.instanceOf(Object),
  paginationWrapper: PropTypes.shape({}).isRequired
};

ViewRoleVacancies.defaultProps = {
  vacancies: []
};

export default PaginatedVacanciesDashboard;
