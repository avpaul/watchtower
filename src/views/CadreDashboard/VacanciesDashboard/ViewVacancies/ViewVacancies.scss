@import '../../../../assets/sass/abstract/variables';
@import '../../../../assets/sass/abstract/mixins';

.ops-vacancies {
  &__wrapper {
    width: 100%;
  }

  &__filter {
    margin-top: 1rem;
    margin-bottom: 1rem;
  }

  &__container {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(232px, 0fr));
    grid-row-gap: 20px;
    grid-column-gap: 5px;
    margin-top: 1rem;
    margin-bottom: 1rem;
    justify-content: space-between;
    min-height: 53vh;
  }
}

.ops-no-vacancies {
  font-size: 24px;
}

.toogle-container {
  display: inline;
}

.vacancy-toggle-button {
  background-color: #ffffff;
  border: 1px solid #0459e4;
  height: 2rem;
  width: 11rem;
  color: #0459e4;
  font-family: $font-primary;
  outline: none !important;
  font: 400 13px system-ui;

  &--active {
    @extend .vacancy-toggle-button;
    color: white;
    background-color: #0459e4;
  }
}

.vacancy-search-div {
  float: right;
  background-color: #fff;
  width: 28%;
  border-radius: 30px;
  border: 1px solid #ffffff;
}

.vacancy-search-input {
  border: none;
  margin-left: 0.5rem;
  padding: 1rem 1.1rem;
  color: #0459e4;
  font-family: 'DIN Pro';
  outline: none;
  height: 2rem;
  text-align: left;

  &::placeholder {
    color: rgba(7, 14, 17, 0.51);
    font-size: 13px;
  }
}

.vacancy-search-icon {
  padding: 1em;
  color: rgba(51, 62, 68, 0.51);
  font-size: 13px;
}
