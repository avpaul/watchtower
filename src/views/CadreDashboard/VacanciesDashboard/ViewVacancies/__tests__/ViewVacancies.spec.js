import React from 'react';
import { shallow } from 'enzyme';
import ViewVacancies from '../ViewVacancies';
import { generateVacancyGroups } from '../../../../../__mocks__/projectVacancy';

describe('Vacancy Dashboard', () => {
  const vacanciesToDisplay = "project";
  const allVacancies = generateVacancyGroups(5, 2);

  /**
   * Creates an enzyme instance to test the VacancyDashboard component.
   *
   * @param propOverrides Used to edit the props passed to the component when being mounted
   * @returns { wrapper, props }
   */
  const setup = (vacancies, toDisplay) => {
    const wrapper = shallow(<ViewVacancies vacancies={vacancies} vacanciesToDisplay={toDisplay} />);
    return { wrapper };
  };

  it('renders correctly', () => {
    const { wrapper } = setup(allVacancies, vacanciesToDisplay);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders no vacancies when array is empty', () => {
    const { wrapper } = setup([]);
    expect(wrapper).toMatchSnapshot();
  });
});
