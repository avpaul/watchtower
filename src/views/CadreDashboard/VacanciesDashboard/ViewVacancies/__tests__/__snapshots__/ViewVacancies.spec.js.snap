// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vacancy Dashboard render paginated view correctly 1`] = `
<PaginationFrontendWrapper
  component={
    <ViewRoleVacancies
      component={
        <ViewRoleVacancies
          paginationWrapper={
            Object {
              "handlePaginationPageChange": [Function],
              "paginateData": [Function],
              "processData": [Function],
              "renderPagination": [Function],
              "state": Object {
                "filteredData": Array [],
                "paginatedData": Array [],
                "paginationFilter": Object {
                  "page": 1,
                  "perPage": 25,
                  "totalPages": 0,
                },
              },
              "updateData": [Function],
            }
          }
          vacancies={
            Object {
              "certificationVacancies": Array [
                Object {
                  "available_slots": 2,
                  "certification": Object {
                    "0": Object {
                      "certification": Object {
                        "description": "Ut aut et incidunt temporibus.",
                        "duration": 20,
                        "exclusive": true,
                        "id": 1,
                        "name": "Lyric Strosin",
                      },
                      "certification_id": 1,
                      "created_at": "2019-07-11 06:54:50",
                      "fellow_id": null,
                      "id": 2,
                      "is_active": false,
                      "updated_at": "2019-07-11 06:54:50",
                    },
                    "description": "Ut aut et incidunt officiis repudiandae temporibus.",
                    "duration": 20,
                    "exclusive": true,
                    "id": 1,
                    "name": "Lyric Strosin",
                  },
                },
              ],
              "projectVacancies": Array [
                Object {
                  "available_slots": 1,
                  "project": Object {
                    "id": 1,
                    "name": "A et.",
                  },
                  "role": Object {
                    "active_engineers": Array [],
                    "description": "Neque aut modi minima dolorem quia itaque.",
                    "id": 6,
                    "name": "FPC",
                  },
                  "vacancies": Array [
                    Object {
                      "fellow_id": null,
                      "id": 1,
                      "is_active": false,
                      "project": Object {
                        "id": 1,
                        "name": "A et.",
                      },
                      "project_id": 1,
                      "project_role_id": 6,
                      "role": Object {
                        "active_engineers": Array [],
                        "description": "Neque aut modi minima dolorem quia itaque.",
                        "id": 6,
                        "name": "FPC",
                      },
                    },
                  ],
                },
              ],
            }
          }
        />
      }
      vacancies={Array []}
    />
  }
/>
`;

exports[`Vacancy Dashboard renders correctly 1`] = `
<div>
  <div>
    <div
      className="toogle-container"
    >
      <button
        className="vacancy-toggle-button--active"
        id="project-button"
        onClick={[Function]}
        type="button"
        value="project"
      >
        Project Vacancies
      </button>
      <button
        className="vacancy-toggle-button"
        id="certification-button"
        onClick={[Function]}
        type="button"
        value="certification"
      >
        Certification Vacancies
      </button>
    </div>
  </div>
  <div
    className="ops-vacancies-container"
  >
    <div
      className="ops-no-vacancies"
    >
      No Vacancies
    </div>
  </div>
  <div
    className=""
  />
</div>
`;

exports[`Vacancy Dashboard renders no vacancies when array is empty 1`] = `
<div>
  <div>
    <div
      className="toogle-container"
    >
      <button
        className="vacancy-toggle-button--active"
        id="project-button"
        onClick={[Function]}
        type="button"
        value="project"
      >
        Project Vacancies
      </button>
      <button
        className="vacancy-toggle-button"
        id="certification-button"
        onClick={[Function]}
        type="button"
        value="certification"
      >
        Certification Vacancies
      </button>
    </div>
  </div>
  <div
    className="ops-vacancies-container"
  >
    <div
      className="ops-no-vacancies"
    >
      No Vacancies
    </div>
  </div>
  <div
    className=""
  />
</div>
`;
